internal/wal_push_handler.go

```diff
@@ -26,10 +26,14 @@ func (err CantOverwriteWalFileError) Error() string {
 // HandleWALPush is invoked to perform wal-g wal-push
 func HandleWALPush(uploader *Uploader, walFilePath string) {
 	uploader.UploadingFolder = uploader.UploadingFolder.GetSubFolder(WalPath)
-	bgUploader := NewBgUploader(walFilePath, int32(getMaxUploadConcurrency(16)-1), uploader)
+	concurrency, err := getMaxUploadConcurrency(16)
+	if err != nil {
+		panic(err)
```

Comment:
You don't need to panic here. Error is in a handler already. So there you can use tracelog.ErrorLogger.Fatalf

Reply:
I just copied it from below :). Your words seem correct, so I also changed it there, after UploadWALFile.